### ISSUE TITLE
[Bug]: [SFT / RoboTwin] PyTorch checkpoint trained with RLinf underperforms native JAX SFT despite similar converged loss

### ISSUE THREAD

**Issue by SunJiayu-L** [RLinf/RLinf#1274]
### Affected component

We are trying to reproduce the RoboTwin adjust_bottle SFT result using the RLinf PyTorch training pipeline, but the resulting checkpoint performs significantly worse than the native OpenPI/JAX checkpoint.

The observed results are:

<img width="589" height="146" alt="Image" src="https://github.com/user-attachments/assets/723eb3c0-7e8e-4e07-8c0f-b683692d8d26" />

### Bug description

We are trying to reproduce the RoboTwin adjust_bottle SFT result using the RLinf PyTorch training pipeline, but the resulting checkpoint performs significantly worse than the native OpenPI/JAX checkpoint.

### Configuration YAML file

defaults:
  - model/pi0_5@actor.model
  - training_backend/fsdp@actor.fsdp_config
  - override hydra/job_logging: stdout

hydra:
  run:
    dir: .
  output_subdir: null
  searchpath:
    - file://${oc.env:EMBODIED_PATH}/config/

cluster:
  num_nodes: 1
  component_placement:
    actor,env,rollout: 0-1

runner:
  task_type: sft
  logger:
    log_path: "../results"
    project_name: rlinf
    experiment_name: "robotwin15_sft_openpi_pi05"
    logger_backends: ["tensorboard"]

  max_epochs: -1
  max_steps: 30000
  val_check_interval: -1
  save_interval: 5000

data:
  train_data_paths: "/mnt/petrelfs/ /jiayusun/store/cache/huggingface/lerobot/robotwin_15_pi05"

actor:
  group_name: "ActorGroup"
  training_backend: "fsdp"
  micro_batch_size: 32
  global_batch_size: 64
  seed: 0

  model:
    precision: null
    model_path: "/mnt/petrelfs/ /jiayusun/pi05_base"
    num_action_chunks: 50
    action_dim: 14
    add_value_head: False
    openpi:
      config_name: "pi05_aloha_robotwin15"
      detach_critic_input: True
      num_images_in_input: 3
      train_expert_only: False

  optim:
    lr: 2.5e-5
    value_lr: 1.55e-4
    adam_beta1: 0.9
    adam_beta2: 0.95
    adam_eps: 1.0e-08
    weight_decay: 1e-10
    clip_grad: 1.0
    total_training_steps: 30000
    lr_warmup_steps: 1000
    lr_scheduler: "cosine"
    num_cycles: 0.5
    min_lr: 2.5e-6

  fsdp_config:
    strategy: "fsdp"
    sharding_strategy: "no_shard"
    use_orig_params: False
    gradient_checkpointing: False
    mixed_precision:
      param_dtype: ${actor.model.precision}
      reduce_dtype: ${actor.model.precision}
      buffer_dtype: ${actor.model.precision}

### Log file

[RLinf_train+eval.txt](https://github.com/user-attachments/files/28922350/RLinf_train%2Beval.txt)
[RLinf_train+Robotwin_eval.txt](https://github.com/user-attachments/files/28922351/RLinf_train%2BRobotwin_eval.txt)
[Robotwin_train+eval.txt](https://github.com/user-attachments/files/28922352/Robotwin_train%2Beval.txt)

### Environment

python -V:
uv pip list:
nvidia-smi:
RLinf version:

[env.txt](https://github.com/user-attachments/files/28922387/env.txt)

Docker image tag: 

### Other reproduction info

here are some ai report: 
<img width="767" height="217" alt="Image" src="https://github.com/user-attachments/assets/8f781db8-bf11-4568-8c7c-9eec411c5c88" />

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

**Comment by qurakchin** [comment]
In fact, openpi's official PyTorch implementation cannot align with the JAX version, and we have internally reproduced this result as well. Based on this issue, we re-implemented a new PyTorch version in RLinf. In our internal tests on SFT tasks, this new implementation has already achieved alignment with the JAX version, and may even deliver better performance. For the code, please refer to #1254.

**Comment by SunJiayu-L** [comment]
Thanks for the clarification.
I checked the RLinf Hugging Face organization and found:    `RLinf/RLinf-Pi05-RoboTwin-SFT-adjust_bottle`
It has already been fine-tuned on the "adjust_bottle" task.   Are there any weights that have not been fine-tuned?

**Comment by qurakchin** [comment]
Sorry, I didn't quite understand your last question. Our RLinf/RLinf-Pi05-RoboTwin-SFT-adjust_bottle was trained from the pi05 base checkpoint using the old RLinf PyTorch version. Are you asking for the pi05 base checkpoint?
